### DISCUSSION TITLE
TuringCourses/major_basic/advanced_data_structure/

### DISCUSSION THREAD

**Discussion by giscus** [ZJU-Turing/TuringCourses#72] [General]
# TuringCourses/major/advanced_data_structure/

图灵班学习指南

https://zju-turing.github.io/TuringCourses/major/advanced_data_structure/

**Comment by HobbitQia** [comment]
ADS 的期末应该是计院学子的梦魇之一了，~~特别是还有个 yds 出题~~。好在这两年叶老师逐渐收手，题目开始更加贴近课内所学，难度有所下降。今年的题目也是如此，叶老师考前宣称：”今年我都没出几道题，你们考得不好不能怪我了。“
今年来说，体感题目整体比去年难，简单题没有这么多，但是编程题把分送到位了（去年 AVL[ac01]），而且没有很难的题，中档题的比例变多了。
关于这门课的期末复习，个人认为可以从这几点去准备：

* 平时听课应该认准张叶毛三人的课，他们是做算法的，本身会补充 PPT 没有的东西。期末最难的题大概率也来自他们之手。比如考前几天很多同学在问 Load Balancing 的近似比，我记得讲近似算法这节课的最后，叶老师提到了这个问题并讲了大致的思路。
* 算法部分，可以看看《算法设计》这本书，上面对于 P&NP、近似算法、局部搜索、随机算法的讲解都非常到位。
* 要把 PPT 过一遍。
    * 数据结构部分的代码要非常熟悉，了解变量的含义。（姥姥的题有可能不会给有效的注释，得凭自己猜）
    * PPT 的重点不是很突出，因此需要细致地看每句话，不要漏掉细节。
    * 部分章节编的不是很好（最差的就是 P&NP 了，上来讲的不知所云…个人认为从多项式归约开始讲会好得多）
* 多做题，这几年的历年卷/练习题资源应该足够丰富了。不会的及时问老师或者助教，因为我感觉不问他们其他人几乎都做不出来，还是得问问专业的人（雾）。
放平心态，不要太紧张。笔者考前模拟题没怎么上过 60，考试的时候感觉题比去年难好多要完蛋了，结果最后考完发现 90 分[ac01]，一瞬间不敢相信这是自己的期末成绩。祝后面的同学好运，~~愿天堂没有 yds 和他出的题~~。

**Comment by MTDickens** [comment]
ADS 诚信测验答案（详见[陈越的官方答案](https://www.zhihu.com/question/68235594/answer/261395370)）：

- 答案为 F: 11, 15, 20, 21, 30, 31, 34, 56
    - 一共 8 道题为 F
- 答案为 T: （其他所有题目）

**Comment by dieKarotte** [comment]
22级的期末参考性不是很大，因为今年大魔王只剩下yds还在，而毛张都没有开课，导致期末考试压力相比往年略有降低，从斩杀情况看也比往年要好，看来大Q老师一年前许下的愿望部分实现了，现在天堂没有张毛的题了，只剩yds和他的题了[ac01]。今年还有ljf老师的神仙助教wyy为每次课程写下了详细的note，对于期末考试可以说是有相当大的帮助，很多小结论作为课上的补充其实就是老师们最喜欢考察的地方，平时课后过一遍wyy的note，在考前再整体过一遍，相信可以查缺补漏，较好地巩固知识点

**Reply by Aquatic-7** [reply]
请问笔者wyy助教的note在哪找哇，我选的是wc老师的课

**Comment by yhwu-is** [comment]
https://yhwu-is.github.io/Teach/tcs/ads/ads/
我把 ADS 的讲义公开在我的 teaching 主页上了，但暂时可能没空很仔细地修订，如果有问题可以联系我
